sys/riotboot/flashwrite.c

```diff
@@ -54,14 +56,36 @@ int riotboot_flashwrite_init_raw(riotboot_flashwrite_t *state, int target_slot,
     state->target_slot = target_slot;
     state->flashpage = flashpage_page((void *)riotboot_slot_get_hdr(target_slot));
 
+    if (CONFIG_RIOTBOOT_FLASHWRITE_RAW && offset) {
```

Comment:
```suggestion
    if (CONFIG_RIOTBOOT_FLASHWRITE_RAW && offset == 0) {
```

We only want to erase the page with the first chunk, no?

Reply:
What I want to do here is erase the first page if an offset is supplied. If no offset is supplied there is apparently no checksum or other mechanism to validate the page and the first page will get automatically erased during the `riotboot_flashwrite_putbytes` function call. If an offset is supplied we have to erase it here because the first page won't be erased during `riotboot_flashwrite_putbytes`, and there is apparently a validation flag to erase.

Do you think this needs to be clarified. I can add either something here or extend the docs in the header files if the overal architecture is not clear.

Reply:
I've pushed a sentence of doc here. Please let me know if it is clear enough :)